### ISSUE TITLE
Failed to load dynamic library

### ISSUE THREAD

**Issue by m-glock** [simolus3/moor#1055]
I have found some other issues with the same problem, but so far none of the proposed answers have helped me. I am new to moor and flutter and followed the 'Getting started' sample code [here](https://moor.simonbinder.eu/docs/getting-started/). Changing the settings.gradle as mentioned [here ](https://github.com/flutter/flutter/issues/55827#issuecomment-623779910) did not help me.

**Error message and call stack:**
```
E/flutter (12873): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library (dlopen failed: library "/data/data/com.example.money_overview/lib/libsqlite3.so" not found)
E/flutter (12873): #0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:55)
E/flutter (12873): #1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:20:12)
E/flutter (12873): #2      _defaultOpen (package:sqlite3/src/load_library.dart:40:31)
E/flutter (12873): #3      OpenDynamicLibrary.openSqlite (package:sqlite3/src/load_library.dart:98:12)
E/flutter (12873): #4      sqlite3 (package:sqlite3/src/api/sqlite3.dart:9:34)
E/flutter (12873): #5      sqlite3 (package:sqlite3/src/api/sqlite3.dart)
E/flutter (12873): #6      _VmDelegate.open (package:moor/src/ffi/vm_database.dart:126:13)
E/flutter (12873): #7      DelegatedDatabase.ensureOpen.<anonymous closure> (package:moor/src/runtime/executor/helpers/engines.dart:255:22)
E/flutter (12873): <asynchronous suspension>
E/flutter (12873): #8      DelegatedDatabase.ensureOpen.<anonymous closure> (package:moor/src/runtime/executor/helpers/engines.dart)
E/flutter (12873): #9      BasicLock.synchronized (package:synchronized/src/basic_lock.dart:32:26)
E/flutter (12873): #10     DelegatedDatabase.ensureOpen (package:moor/src/runtime/executor/helpers/engines.dart:249:25)
E/flutter (12873): #11     LazyDatabase.ensureOpen.<anonymous closure> (package:moor/src/utils/lazy_database.dart:43:49)
E/flutter (12873): #12     _rootRunUnary (dart:async/zone.dart:1198:47)
E/flutter (12873): #13     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
E/flutter (12873): #14     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
E/flutter (12873): #15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
E/flutter (12873): #16     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
E/flutter (12873): #17     Future._completeWithValue (dart:async/future_impl.dart:529:5)
E/flutter (12873): #18     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
E/flutter (12873): #19     _rootRun (dart:async/zone.dart:1190:13)
E/flutter (12873): #20     _CustomZone.run (dart:async/zone.dart:1093:19)
E/flutter (12873): #21     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
E/flutter (12873): #22     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
E/flutter (12873): #23     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
E/flutter (12873): #24     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
E/flutter (12873): 
```

**Database class:**
```
import 'package:moor/ffi.dart';
import 'package:moor/moor.dart';
import 'package:path_provider/path_provider.dart';
import 'package:path/path.dart' as path;
import 'dart:io';

part 'Database.g.dart';

// flutter packages pub run build_runner build
LazyDatabase _openConnection() {
  return LazyDatabase(() async {
    final dbFolder = await getApplicationDocumentsDirectory();
    final file = File(path.join(dbFolder.path, 'db.sqlite'));
    return VmDatabase(file);
  });
}

@UseMoor(
    tables: [Categories, Expanses, Preferences, MonthlyExpanses, SavingGoals, Savings]
)
class Database extends _$Database {
  // we tell the database where to store the data with this constructor
  Database() : super(_openConnection());

  // bump this number whenever you change or add a table definition
  @override
  int get schemaVersion => 4;
}
```

**pubspec.yaml:**
```
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  moor: 3.4.0
  sqlite3_flutter_libs: 0.4.0
  path_provider: 1.6.27

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.0

dev_dependencies:
  flutter_test:
    sdk: flutter
  moor_generator: 3.4.1
  build_runner: 1.11.1
```

**Comment by simolus3** [comment]
Thanks for the report. Unfortunately I messed up when releasing `0.4.0` of `sqlite3_flutter_libs`. I released `0.4.0+1` shortly after, can you try that one?

```yaml
dependencies:
  sqlite3_flutter_libs: ^0.4.0+1
```

In general, I suggest using the carret syntax for dependencies so that pub can try to give you the latest version. Apologies for the inconvenience caused by `0.4.0`.

**Comment by m-glock** [comment]
Thank you for the quick reply, it works now!
